tfhe/src/integer/server_key/radix/sub.rs

```diff
@@ -420,124 +419,23 @@ impl ServerKey {
-                    )
-                },
-            );
-
-            for block in result.blocks[1..num_blocks - 1].iter_mut() {
-                input_carry = block_add_assign_returning_carry(self, block, &input_carry);
-            }
-
-            // Treat the last block separately to handle last step of overflow behavior
-            let output_carry = block_add_assign_returning_carry(
-                self,
-                &mut result.blocks[num_blocks - 1],
-                &input_carry,
-            );
-            let overflowed = self.resolve_signed_overflow(
-                last_block_inner_propagation,
-                &BooleanBlock::new_unchecked(input_carry),
-                &BooleanBlock::new_unchecked(output_carry),
-            );
-
-            return (result, overflowed);
-        }
-
-        // 1_X parameters
-        //
-        // Same idea as other algorithms, however since we have 1 bit per block
-        // we do not have to resolve any inner propagation but it adds one more
-        // sequential PBS
-        if self.key.message_modulus.0 == 2 {
-            if signed_operation == SignedOperation::Subtraction {
-                self.unchecked_sub_assign(&mut result, rhs);
-            } else {
-                self.unchecked_add_assign(&mut result, rhs);
-            }
-
-            let mut input_carry = self.key.create_trivial(0);
-            for block in result.blocks[..num_blocks - 1].iter_mut() {
-                input_carry = block_add_assign_returning_carry(self, block, &input_carry);
-            }
-
-            let output_carry = block_add_assign_returning_carry(
-                self,
-                &mut result.blocks[num_blocks - 1],
-                &input_carry,
-            );
-
-            // Encode the rule
-            // "Overflow occurred if the carry into the last bit is different than the carry out
-            // of the last bit"
-            let overflowed = self.key.not_equal(&output_carry, &input_carry);
-            return (result, BooleanBlock::new_unchecked(overflowed));
-        }
-
-        panic!(
-            "Invalid combo of message modulus ({}) and carry modulus ({}) \n\
-            This function requires the message modulus >= 2 and carry modulus >= message_modulus \n\
-            I.e. PARAM_MESSAGE_X_CARRY_Y where X >= 1 and Y >= X.",
-            self.key.message_modulus.0, self.key.carry_modulus.0
-        );
-    }
     pub fn unchecked_signed_overflowing_sub(
         &self,
         lhs: &SignedRadixCiphertext,
         rhs: &SignedRadixCiphertext,
     ) -> (SignedRadixCiphertext, BooleanBlock) {
-        self.unchecked_signed_overflowing_add_or_sub(lhs, rhs, SignedOperation::Subtraction)
+        let flipped_rhs = self.bitnot(rhs);
+        let carry = self.create_trivial_boolean_block(true);
+        let mut result = lhs.clone();
+        let overflowed = self
+            .advanced_add_assign_with_carry_sequential_parallelized(
```

Comment:
parallelized algo in sequential non parallel code ?

Btw should we get rid of the non parallel code ?

Reply:
The parallel sequential only uses 2 threads at a time so its okayish

And yes I don't think non parallel code provide much value